common/thorhelper/thorcommon.hpp

```diff
@@ -281,15 +281,29 @@ class ActivityTimer
         else
             startCycles = 0;
     }
-
-    ~ActivityTimer()
+    inline void flushTimes(cycle_t nowCycles)
+    {
+        accumulator.endCycles = nowCycles;
+        accumulator.totalCycles += (nowCycles - startCycles);
+    }
+    inline void flushTimesIntermittantly()
     {
         if (enabled)
         {
             cycle_t nowCycles = get_cycles_now();
-            accumulator.endCycles = nowCycles;
-            cycle_t elapsedCycles = nowCycles - startCycles;
-            accumulator.totalCycles += elapsedCycles;
+            if (nowCycles - (cycle_t) startCycles > defaultFlushTimerInterval)
+            {
+                flushTimes(nowCycles);
+                startCycles = nowCycles;
```

Comment:
To improve the code, we could store the subtraction to be used in both the comparison and assignment later. We could eliminate flushTimes and put that code in flushTimesIntermittantly, if we were not sure that compiler would handle the inline specifier as efficiently.

As far as the suggestion to eliminate the comparison altogether, it will mean with every call to flushTimesIntermittantly, we would have:
1) extra work in the way of 2 assignments (accumulator.endCycles and startCycles) and 1 addition (calculate new value for totalCycles)
3) but save a little time by not needing to do the comparison.

It would be simpler but I would think it will be a little slower especially.

Reply:
probably slower to remove the condition/always assign vs calc. diff once + condition.
Might be worth timing running each in a tight loop.

But a simple counter as in other comment might be best, because you really want to try to do as close to nothing if possible in vast majority of calls given how many billions of times this will be called.

@ghalliday - what do you think of a self-adjusting countdown counter ?